### ISSUE TITLE
HybridProvider support for MessagePack

### ISSUE THREAD

**Issue by jasenf** [dotnetcore/EasyCaching#270]
Hi,

Is the Redis Hybrid provider using the configured serializer?  We upgraded to .Net 5, had to switch out the BinaryFormatter but even after reconfiguring we get still get 

System.NotSupportedException: BinaryFormatter serialization and deserialization are disabled within this application. See https://aka.ms/binaryformatter for more information.
   at System.Runtime.Serialization.Formatters.Binary.BinaryFormatter.Serialize(Stream serializationStream, Object graph)
   at EasyCaching.Core.Serialization.DefaultBinaryFormatterSerializer.Serialize[T](T value)
   at EasyCaching.Redis.DefaultRedisCachingProvider.BaseSet[T](String cacheKey, T cacheValue, TimeSpan expiration)
   at EasyCaching.Core.EasyCachingAbstractProvider.Set[T](String cacheKey, T cacheValue, TimeSpan expiration)
   at EasyCaching.HybridCache.HybridCachingProvider.Set[T](String cacheKey, T cacheValue, TimeSpan expiration)

I don't see any references to support for different serializer in the HybrdProvider or the RedisBus providers.

Is this the case?

thanks!

**Comment by catcherwong** [comment]
@jasenf  Thanks for your interest in this project.

BinaryFormatter is the default serializer of `IEasyCachingBus`. 

https://github.com/dotnetcore/EasyCaching/blob/61f1bd77a822dca7c1e8ee29249000482aa146f2/bus/EasyCaching.Bus.Redis/Configurations/RedisBusOptionsExtension.cs#L38

You can add your own serializer to replace it as well. 

```cs
services.AddSingleton<IEasyCachingSerializer, OwnSerializer>();
```

This solution can solve your current problem, but it is not particularly good.  It might be better that working the same with the caching provider.

**Comment by jasenf** [comment]
Is there any plan to add a Moe normalized way to do this, like .WithMessagePack()?

**Comment by catcherwong** [comment]
Yes, This is already planned.

**Comment by catcherwong** [comment]
We released `v1.2.0-alpha20201216055857` at MyGet to improve this issue.

You can have a try with this version. The MyGet url is https://www.myget.org/F/ncc/api/v3/index.json
